drivers/interrupt_controller/intc_vim.c

```diff
@@ -0,0 +1,140 @@
+/* Copyright (C) 2023 BeagleBoard.org Foundation
+ * Copyright (C) 2023 S Prashanth
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/arch/cpu.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/arch/arm/aarch32/irq.h>
+
+#include <zephyr/drivers/interrupt_controller/intc_vim.h>
+
+unsigned int z_soc_irq_get_active(void)
```

Comment:
You are polluting the `z_soc_*` namespace. Please check how this done for other interrupt controllers.